### PR TITLE
Restructuring and extended wiper signals.

### PR DESCRIPTION
**Pull request by erikbosch**
This PR adds a large number of signals for detailed control of wipers. 

The major contribution is part of a "System" branch. It intends to represent an advanced wiper system where each window (front and rear) can be served by up to two wiper systems (master and slave). Most vehicles of today have only one wiper system per window, even if it has multiple wipers they are mechanically connected and cannot move independently. Multiple wiper systems are needed if the wipers are controlled individually and can move independently. In that case the "slave" wiper system needs to consider movement of the "master" wiper system to avoid that the wipers crash into each other.

Having "System" in a separate branch/file makes it easy for an OEM to delete or replace the wiper "System" signals if they would use or need other signals for low level control.

### CODE REVIEW

**Comment by erikbosch** [comment]
Concerning separation on user-level vs. system-level, one aspect for the division is that a windshield might have multiple more or less independent wiper-systems (e.g. multiple motors rather than a single motor controlling multiple mechanically linked wipers), that is the reason for proposing `master` and `slave` sub-branches under the `SystemControl` branch.

Current proposal:  
```

Vehicle.Body.Windshield.Front.Wiping.UserControl.Mode
Vehicle.Body.Windshield.Rear.Wiping.UserControl.Mode

Vehicle.Body.Windshield.Front.Wiping.SystemControl.Master.Mode
Vehicle.Body.Windshield.Front.Wiping.SystemControl.Slave.Mode

Vehicle.Body.Windshield.Rear.Wiping.SystemControl.Master.Mode
Vehicle.Body.Windshield.Rear.Wiping.SystemControl.Slave.Mode
```

... so if we are to support multiple systems for a single windshield we need some form of separation, but this could of course be generated differently, possibly like below:

```
# Common wiper signals for the front windshield
Vehicle.Body.Windshield.Front.Wiping.Mode
# Wiper signals valid only for the master wiping system for the front windshield
Vehicle.Body.Windshield.Front.Wiping.System.Master.Mode
# Wiper signals valid only for the slave wiping system for the front windshield
Vehicle.Body.Windshield.Front.Wiping.System.Slave.Mode

```

**Comment by danielwilms** [comment]
@danielwilms will look into it

**Comment by danielwilms** [comment]
Some comments from my side:
1. Branches `system` and `service` I think are unnecessary as they describe nodes, which can go directly under wiper and don't bring any additional context.
2. The separation `master`, `slave` I would see rather as implementation detail and not in the context of the data model. And as you said it's not very common and then we can instantiate the main branch directly in a deployment. Though, if we decide for it, I'd rather use `primary`/`main` and `secondary` instead of `master` & `slave`

**Comment by danielwilms** [comment]
meeting 2022/05/24:
- change master/slave
- think about renaming system, or if we leave it as is

**Comment by erikbosch** [comment]
Pr updated. Service branch removed. System-branch kept, but instances renamed to primary and secondary. Reason to keeping "system" and instances is that "primary" and "secondary" is not just a logical division, it could for example be a big primary wiper moving slowly and a second smaller wiper moving much faster, so if you want to know the frequency of both you must ask for both datapoints.

Concerning variation on car architecture, i.e. whether they have 1 or 2 systems, I can see some different deployment alternatives. If we, like in the current PR version, add instances for primary and secondary, then a vehicle with only one system must use the primary-path,  or alternatively remove the instance information by the layer-concept. Alternatively, if we consider vehicles with double system (primary+secondary) as very rare, we could remove the instances-part from the default-version and instead add a comment that for double-systems instance handling shall be added in a layer.

**Comment by danielwilms** [comment]
> Alternatively, if we consider vehicles with double system (primary+secondary) as very rare, we could remove the instances-part from the default-version and instead add a comment that for double-systems instance handling shall be added in a layer.

As they're rare indeed, I'd vote for the comment.

File: spec/Body/Body.vspec [hunk 1]
```diff
@@ -87,16 +87,57 @@ Windshield:
...
+  description: Relative intensity/sensitivity for interval and rain sensor mode as requested by user/driver.
+               Has no significance if Windshield.Wiping.Mode is OFF/SLOW/MEDIUM/FAST
+               0 - wipers inactive.
+               1 - minimum intensity (lowest frequency/sensitivity, longest interval).
+               2/3/4/... - higher intensity (higher frequency/sensitivity, shorter interval).
+               Maximum value supported is vehicle specific.
+
+Windshield.Wiping.System:
+  type: branch
+#  instances: [Primary, Secondary]
```
[the remark this answers is not in the document]
Comment (danielwilms) [reply spec/Body/Body.vspec:115]: Thanks @erikbosch, will merge it as of the discussion yesterday.

**Pull request merged**